appsec/src/helper/client.cpp

```diff
@@ -211,104 +211,34 @@ bool client::handle_command(const network::client_init::request &command)
     return !has_errors;
 }
 
-bool client::handle_command(network::request_init::request &command)
-{
-    if (!service_) {
-        // This implies a failed client_init, we can't continue.
-        SPDLOG_DEBUG("no service available on request_init");
-        send_error_response(*broker_);
-        return false;
-    }
-
-    if (!compute_client_status()) {
-        auto response_cf =
-            std::make_shared<network::config_features::response>();
-        response_cf->enabled = false;
-
-        SPDLOG_DEBUG("sending config_features to request_init");
-        try {
-            return broker_->send(response_cf);
-        } catch (std::exception &e) {
-            SPDLOG_ERROR(e.what());
-        }
-
-        return true;
-    }
-
-    // During request init we initialize the engine context
-    context_.emplace(*service_->get_engine());
-
-    SPDLOG_DEBUG("received command request_init");
-
-    auto response = std::make_shared<network::request_init::response>();
-    try {
-        auto res = context_->publish(std::move(command.data));
-        if (res) {
-            switch (res->type) {
-            case engine::action_type::block:
-                response->verdict = network::verdict::block;
-                response->parameters = std::move(res->parameters);
-                break;
-            case engine::action_type::redirect:
-                response->verdict = network::verdict::redirect;
-                response->parameters = std::move(res->parameters);
-                break;
-            case engine::action_type::record:
-            default:
-                response->verdict = network::verdict::record;
-                response->parameters = {};
-                break;
-            }
-
-            response->triggers = std::move(res->events);
-            response->force_keep = res->force_keep;
-
-            DD_STDLOG(DD_STDLOG_ATTACK_DETECTED);
-        } else {
-            response->verdict = network::verdict::ok;
-        }
-    } catch (const invalid_object &e) {
-        // This error indicates some issue in either the communication with
-        // the client, incompatible versions or malicious client.
-        SPDLOG_ERROR("invalid data format provided by the client");
-        send_error_response(*broker_);
-        return false;
-    } catch (const std::exception &e) {
-        // Uncertain what the issue is... lets be cautious
-        DD_STDLOG(DD_STDLOG_REQUEST_ANALYSIS_FAILED, e.what());
-        send_error_response(*broker_);
-        return false;
-    }
-
-    SPDLOG_DEBUG(
-        "sending response to request_init, verdict: {}", response->verdict);
-    try {
-        return broker_->send(response);
-    } catch (std::exception &e) {
-        SPDLOG_ERROR(e.what());
-    }
-
-    return false;
-}
-
-bool client::handle_command(network::request_exec::request &command)
+bool client::emplace_service(const std::string &name)
```

Comment:
Any suggestion to get rid of this parameter would welcome. It's only needed for logging purposes

Reply:
Thanks done!